src/dnf-plugins/product-id.py

```diff
@@ -82,16 +89,20 @@ def _download_productid(self, repo, tmpdir):
     def get_enabled(self):
         """find repos that are enabled"""
         lst = []
-        enabled = self.base.repos.iter_enabled()
+
+        # Re-read all repositories, because 'dnf remove' marks all repositories disabled
+        self.base.reset(repos=True)
+        self.base.read_all_repos()
```

Comment:
This two lines will break dnf completely. Please do not do that.

Reply:
Please try to figure out how to use available resources during the run. Not to try to destroy them. The operation will result:
1. Enabling disabled repos from commandline
2. Reset all repos setting changed from commandlines
3. Disable repositories enabled from commandlines.
Probably many other things

Reply:
@j-mracek Do you have a recommended way to get the currently enabled repositories when 'dnf remove' is being run (per comment on line 93)?

Reply:
All available repos are disabled during fill_sack if () if load_available_repos=False. load_available_repos=False because demands.available_repos = false. When command sets demands it cannot be overwritten.
What about to ask another question - Do you really need that information even if the command performing the transaction doesn't require it? I believe that you gathered information about available repositories and packages already in previous transaction. Then I have only removal operation from system, therefore I could calculate a diff with good assumption of exact status on monitored system. I believe that you know much better than me how to optimize the workflow. Re-download and resend of all information is bulletproof approach but not efficient one.

Reply:
All available repos can be seen during config phase.
Add in class ProductId(dnf.Plugin):
```
    def config(self):
        for x in self.base.repos.iter_enabled():
            print("config", x.name)
```

Reply:
@j-mracek Thanks for hint with `config()` hook. It works as we expect.